### PR TITLE
AI's can now play sounds to the station.

### PR DESCRIPTION
**Pull request by ghost**
Oh yes, I did.
Three things to note:
- BYOND has a natural limit on file size or something of what you can upload.
- AI and admins have a way to stop playback.
- Oh yes, I did.

:cl:
 * rscadd: AI's can now play sounds from your computer. If you abuse this, we will find you.

### CODE REVIEW

**Comment by Tokanova** [comment]
so AI can be a DJ now?

**Comment by Tokanova** [comment]
also what's stopping people from uploading and playing a sound file of themselves saying the N word over and over

**Comment by ghost** [comment]
self respect and threat of being lynched, if the AI does some annoying shit you have all rights to kill them
also, we'll probably need some rules set in place since unlike printing library porn, this forces itself into the ears of all people on your z level

**Comment by plosky1** [comment]
If I can't get +sound, then sure as hell AI's can't either.

**Comment by utilitas-** [comment]
\>AI becomes the micspam role
I will be okay with this if the librarian also gets a station-wide meme broadcaster

**Comment by brndd** [comment]
Just make the existing AI announcement thing better by adding more sounds and making it easier to browse the ones that exist. This will just result in ear rape.

**Comment by ghost** [comment]
@plosky1 what if I were to sneakycode a change to phase out +sound and merge it with full admin perms

**Comment by boy2mantwicethefam** [comment]
Ah, SCP:SL how I've missed you
We should merge this at least for a while so I can play Alabama N-word over to the station

**Comment by Tokanova** [comment]
Make sure it can only play sound files. Don't want anyone using this to send out viruses or some shit

**Comment by PaulRitter** [comment]
> Make sure it can only play sound files. Don't want anyone using this to send out viruses or some shit

this.

also maybe add a way for admins to permanently forbid someone from uploading sounds

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -206,6 +206,56 @@ var/list/ai_list = list()
...
+	set name = "Play Sound to Station"
+	
+	if(isUnconscious())
+		return
+		
+	if(ai_sound_ready_time > world.time)
+		to_chat(src, "<span class='notice'>Please wait [round((ai_sound_ready_time - world.time) / 10)] seconds.</span>")
+		return
+	
+	var/sound/uploaded_sound = sound(S, repeat = 0, wait = 1, channel = VOX_CHANNEL, volume = 60)
```
Comment (PaulRitter) [inline code/modules/mob/living/silicon/ai/ai.dm:223]: why `VOX_CHANNEL`?

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -206,6 +206,56 @@ var/list/ai_list = list()
...
+	var/prompt = alert(src, "Are you sure you want to play this sound?","Don't abuse it!","Yes","No")
+	if(prompt == "No")
+		return
+	log_admin("Player [key_name(src)] played sound [S] as an AI.")
+	message_admins("Player [key_name(src)] played sound [S] as an AI.", 1)
+	for(var/mob/M in player_list)
+		if(!M.client)
+			continue
+		if(M.client.prefs.toggles & SOUND_MIDI && M.z == z)
+			M << uploaded_sound
```
Comment (PaulRitter) [inline code/modules/mob/living/silicon/ai/ai.dm:236]: at least `istype(uploaded_sound)` yeez

Comment (PaulRitter) [reply code/modules/mob/living/silicon/ai/ai.dm:236]: *istype S

**Comment by jknpj** [comment]
Letting anyone use +SOUND is the dumbest idea i've ever seen coded. Especially when you can make 9+ AIs easily. Literally just alllow them to use the admin button that add musics to jukebox, at least it is less worse.

**Comment by calzilla1** [comment]
Maybe make it a white list

**Comment by ShiftyRail** [comment]
Is this a clever plan to get AIs banned/removed

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by Irradiation** [comment]
Why yes, let's give an admin permission that takes a while to obtain as an admin to every single AI player. What could go wrong?

**Comment by bathosbathtime** [comment]
Give the research director a remote control to stop the AI's current and/or future sound files and this PR is golden

**Comment by Irradiation** [comment]
Better yet, give the Research Director a button to destroy the AI and mute every sound coming from it.

**Comment by SonixApache** [comment]
this is idiotic

**Comment by ShiftyRail** [comment]
Much like Sonix said, this is dumb
Unlike a meme like getting castle merged for one round, this could be actually a big annoyance on Pomf's end and could be dangerous for players

In the eternal tongue of Molière : "Non"

**Pull request closed without merging**